test/e2e/karmadactl_test.go

```diff
@@ -0,0 +1,330 @@
+	"github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/karmada-io/karmada/pkg/karmadactl"
+	"github.com/karmada-io/karmada/pkg/karmadactl/options"
+	"github.com/karmada-io/karmada/test/e2e/framework"
+	"github.com/karmada-io/karmada/test/helper"
+)
+
+var _ = ginkgo.Describe("Karmadactl promote testing", func() {
+	var karmadaConfig karmadactl.KarmadaConfig
+	var member1 string
+	var member1Client kubernetes.Interface
+
+	ginkgo.BeforeEach(func() {
+		karmadaConfig = karmadactl.NewKarmadaConfig(clientcmd.NewDefaultPathOptions())
+		member1 = framework.ClusterNames()[0]
+		member1Client = framework.GetClusterClient(member1)
+	})
+
+	// test namespaced resource deployment with cluster resource namespace
+	ginkgo.Context("Test promoting namespaced resource: deployment", func() {
+		var deployment *appsv1.Deployment
+		var deploymentNamespace, deploymentName string
+
+		ginkgo.BeforeEach(func() {
+			karmadaConfig = karmadactl.NewKarmadaConfig(clientcmd.NewDefaultPathOptions())
+
+			deploymentNamespace = fmt.Sprintf("karmadatest-%s", rand.String(RandomStrLength))
+			deploymentName = deploymentNamePrefix + rand.String(RandomStrLength)
+			deployment = helper.NewDeployment(deploymentNamespace, deploymentName)
+		})
+
+		ginkgo.AfterEach(func() {
+			framework.RemoveDeployment(kubeClient, deploymentNamespace, deploymentName)
+			framework.RemovePropagationPolicy(karmadaClient, deploymentNamespace, deploymentName+"-propagation")
+			framework.RemoveNamespace(kubeClient, deploymentNamespace)
+		})
+
+		ginkgo.It("Test promoting a deployment from cluster member", func() {
+
+			//  Step 1,  create ns and deploy on cluster member1
+			ginkgo.By(fmt.Sprintf("Creating deployment %s with namespace %s not existed in karmada control plane", deploymentName, deploymentNamespace), func() {
+				deploymentNamespaceObj := helper.NewNamespace(deploymentNamespace)
+				framework.CreateNamespace(member1Client, deploymentNamespaceObj)
+				framework.CreateDeployment(member1Client, deployment)
+			})
+
+			//  Step 2,  promote deploy from cluster member1 to karmada
+			ginkgo.By(fmt.Sprintf("Promoting deployment %s from member: %s to karmada control plane", deploymentName, member1), func() {
+				opts := karmadactl.CommandPromoteOption{
+					Namespace:        deploymentNamespace,
+					Cluster:          member1,
+					ClusterNamespace: options.DefaultKarmadaClusterNamespace,
+				}
+				args := []string{"deployment", deploymentName}
+				//init args: place deployment name to CommandPromoteOption.name
+				err := opts.Complete(args)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+				//use karmadactl to promote a deployment from member1 and it will failed because the ns is not existed
+				err = karmadactl.RunPromote(os.Stdout, karmadaConfig, opts, args)
+				gomega.Expect(err).Should(gomega.HaveOccurred())
```

Comment:
Error info can be varied, can you judge it more specifically? Alternatively, we may not test this error scenario.

Reply:
K8S api server will return a not found error but karmadactl convert it to normal .  I can remove it here